aea/aea_builder.py

```diff
@@ -1644,6 +1645,33 @@ def _check_pypi_dependencies(self, configuration: ComponentConfiguration) -> Non
                     f"Conflict on package {pkg_name}: specifier set '{dep_info.version}' not satisfiable."
                 )
 
+    @staticmethod
+    def check_project_dependencies(
+        agent_configuration: AgentConfig, project_path: Path
+    ) -> None:
+        """Check project config for missing dependencies."""
+
+        dep_tree: Set[ComponentId] = set()
+        for level in DependencyTree.generate(project_path):
+            dep_tree.update(
+                {
+                    ComponentId(package.package_type.value, package.public_id)
+                    for package in level
+                    if package.package_type != PackageType.AGENT
```

Comment:
It's impossible to be a package type agent. If it was, we should raise a ValueError imo as it indicates a problem in our code.

Reply:
So the dependency tree generator scans for all available packages in the project dir which means it'll also count in the agent package so we have to filter it out.